reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -0,0 +1,262 @@
+	 *     <li>Without {@link Subscriber}: all elements pushed to this sink are remembered and will
+	 *     be replayed once the {@link Subscriber} subscribes.</li>
+	 * </ul>
+	 */
+	@SuppressWarnings("deprecation")
+	public static <T> StandaloneFluxSink<T> unicast() {
+		return new FluxProcessorSink<>(UnicastProcessor.create());
+	}
+
+	/**
+	 * A {@link StandaloneMonoSink} that can be triggered at any time by either of
+	 * the three completions: {@link ScalarSink#success(Object) valued completion},
+	 * {@link ScalarSink#success() empty completions} or {@link ScalarSink#error(Throwable) error}.
+	 * This completion is replayed to late subscribers.
+	 */
+	public static <T> StandaloneMonoSink<T> trigger() {
+		return new MonoProcessorSink<>(MonoProcessor.create());
+	}
+
+	// == interfaces ==
+
+	/**
+	 * A flavor of {@link SequenceSink} that is not attached to a single {@link Subscriber}
+	 * but rather viewable {@link #asFlux() as a Flux}. Most likely, such a {@link SequenceSink}
+	 * is capable of multicasting to several subscribers. However implementations can chose
+	 * to be unicast instead, and only accept one {@link Subscriber} at a time (as allowed
+	 * by the {@link org.reactivestreams.Publisher} specification).
+	 *
+	 * @param <T> the type of elements that can be emitted through this sink
+	 */
+	public interface StandaloneFluxSink<T> extends SequenceSink<T> {
+
+		@Override
+		StandaloneFluxSink<T> next(T t);
+
+		/**
+		 * Return the companion {@link Flux} instance that is backed by this sink.
+		 * All calls to this method return the same instance.
+		 *
+		 * @return the {@link Flux} view associated to this {@link StandaloneFluxSink}
+		 */
+		Flux<T> asFlux();
+
+	}
+
+	/**
+	 * A flavor of {@link ScalarSink} that is not attached to a single {@link Subscriber}
+	 * but rather viewable {@link #asMono() as a Mono}. Most likely, such a {@link ScalarSink}
+	 * is capable of multicasting to several subscribers. However implementations can chose
+	 * to be unicast instead, and only accept one {@link Subscriber} at a time (as allowed
+	 * by the {@link org.reactivestreams.Publisher} specification).
+	 *
+	 * @param <T> the type of elements that can be emitted through this sink
+	 */
+	public interface StandaloneMonoSink<T> extends ScalarSink<T> {
+
+		/**
+		 * Return the companion {@link Mono} instance that is backed by this sink.
+		 * All calls to this method return the same instance.
+		 *
+		 * @return the {@link Mono} view associated to this {@link StandaloneMonoSink}
+		 */
+		Mono<T> asMono();
+	}
+
+	// == concrete classes ==
+
+	static final class FluxProcessorSink<T>
+			implements StandaloneFluxSink<T> {
+
+		final FluxSink<T>         delegateSink;
+		final FluxProcessor<T, T> processor;
+
+		@SuppressWarnings("deprecation")
+		FluxProcessorSink(FluxProcessor<T, T> processor) {
+			this.processor = processor;
+			this.delegateSink = processor.sink();
+		}
+
+		@Override
+		public Flux<T> asFlux() {
+			return processor;
```

Comment:
Since `asFlux()` exposes `Flux<T>` but not concrete `Processor` `onEachOperator` hook could be applied here.  
Since `FluxProcessor`s are `ASYNC` `Publisher`s it could help with `Thread` context propagations

Reply:
the processor is only fed via manual interactions, so I'm not sure how that would help eg. if the user decides to do a `new Thread(() -> sink.next(1)).start()`...

Reply:
it would help downstream consumers as processor could be decorated and values would be emitted within context for them.  
here the example of the problem   
```java
//add hook that propagates value from Reactor `Context` to the `ThreadLocal`
....
String testValue = "testValue";
setTL(testValue);
MonoProcessor<Integer> processor = MonoProcessor.create();
Mono.delay(Duration.ofMillis(500))
        .doOnNext(it -> processor.onNext(1))
        .subscribe();

String value = processor
//        .doOnNext(it -> {}) // works only with dummy operator as it decorates downstream
        .map(it -> getValueFromTL())// value was not propagated because hook was not applied to the source
        .block();
Assertions.assertThat(value).isEqualTo(testValue);
 
```

Reply:
I need to look deeper into the consequences of that, not sure applying the hooks to the processor will work naturally. plus currently there is no such thing with processors. can you chime back in once M1 is released (eg creating an issue for M2) ?